### ISSUE TITLE
Incorrect array key values reported in the Watch, Locals and Globals debugger panels

### ISSUE THREAD

**Issue by WheelDogs** [Komodo/KomodoEdit#2316]
The key values of arrays as reported while debugging are incorrect. Using parray to display the same array presents the correct (actual) key values as expected in the Output panel.

### Steps to Reproduce
Create an array with these keys...

### Expected results
::SP_TTS
   voices
               CereVoice Adam - English (East Coast America)        string ::tcom::handle0x023EAEA8
               CereVoice Katherine - English (East Coast America   string ::tcom::handle0x023EACC8
               Microsoft David Desktop - English (United States)     string ::tcom::handle0x023EAE08
etc

### Actual results
   voices  
             East Coast America)        string ::tcom::handle0x023EAEA8
             East Coast America)        string ::tcom::handle0x023EACC8
             United States)                 string ::tcom::handle0x023EAE08

### Platform Information

*Komodo IDE
*Komodo Version 10,1,4
*Operating System (and version) Win10, Win7

### Additional Information
This is fundamental stuff...

 
![arrayerror](https://cloud.githubusercontent.com/assets/21134598/22667879/a013e38e-ec8c-11e6-8572-94042acbaf7b.png)

**Comment by mitchell-as** [comment]
Please post some sample code that exhibits this error so I can reproduce it on my machine in order to debug and fix. Thank you.

**Comment by WheelDogs** [comment]
array set voices [list \
{Cepstral Allison} {Cepstral Allison} \
{Cepstral William}   {Cepstral William }                \
{CereVoice Adam - English (East Coast America)}       {CereVoice Adam - English (East Coast America)}  \
{CereVoice Katherine - English (East Coast America)}  {CereVoice Katherine - English (East Coast America)}  \
{IVONA 2 Eric - US English male voice [22kHz]}        {IVONA 2 Eric - US English male voice [22kHz]}  \
{IVONA 2 Jennifer - US English female voice [22kHz]}  {IVONA 2 Jennifer - US English female voice [22kHz]}  \
{IVONA 2 Joey - US English male voice [22kHz]}        {IVONA 2 Kimberly - US English female voice [22kHz]}  \
{IVONA 2 Salli - US English female voice [22kHz]}     {IVONA 2 Salli - US English female voice [22kHz]}  \
{Microsoft David Desktop - English (United States)}   {Microsoft David Desktop - English (United States)}  \
{VW Julie}                                            {VW Julie}  \
{VW Kate}                                            {VW Kate}  \
Zira                                                 {Microsoft Zira Desktop - English (United States)}]

**Comment by mitchell-as** [comment]
I tried to put that in a sample Tcl script, but I get the following error:

```Error in startup script: invalid command name "Cepstral Allison"invalid command name "Cepstral Allison"
    while executing
"{Cepstral Allison} {Cepstral Allison}"
    invoked from within
"DbgNub_uplevelCmd 1 $cmd"
    invoked from within
"DbgNub_uplevelCmd 1 $icode"
    (procedure "source" line 1)
    invoked from within
"source $argv0"
    invoked from within
"if {$encoding eq {}} {
    source $argv0
} else {
    source -encoding $encoding $argv0
}"```

I know next to nothing about Tcl, so please advise or provide a full, working script. Thank you.

**Comment by WheelDogs** [comment]
Well, that makes things even more interesting.

I just executed it (again) using Komodo...and no error. (after all, the intent is to debug Komodo)

Ran it using tkcon and it produces the above error...

**Comment by WheelDogs** [comment]
Seems there a two issues with Komodo..

**Comment by WheelDogs** [comment]
The problem with the code I sent was that the paste into GitHub removed the trailing backslashes.

**Comment by WheelDogs** [comment]
array set voices [list \
{Cepstral Allison} {Cepstral Allison} \
{Cepstral William}   {Cepstral William }                \
{CereVoice Adam - English (East Coast America)}       {CereVoice Adam - English (East Coast America)}  \
{CereVoice Katherine - English (East Coast America)}  {CereVoice Katherine - English (East Coast America)}  \
{IVONA 2 Eric - US English male voice [22kHz]}        {IVONA 2 Eric - US English male voice [22kHz]}  \
{IVONA 2 Jennifer - US English female voice [22kHz]}  {IVONA 2 Jennifer - US English female voice [22kHz]}  \
{IVONA 2 Joey - US English male voice [22kHz]}        {IVONA 2 Kimberly - US English female voice [22kHz]}  \
{IVONA 2 Salli - US English female voice [22kHz]}     {IVONA 2 Salli - US English female voice [22kHz]}  \
{Microsoft David Desktop - English (United States)}   {Microsoft David Desktop - English (United States)}  \
{VW Julie}                                            {VW Julie}  \
{VW Kate}                                            {VW Kate}  \
Zira                                                 {Microsoft Zira Desktop - English (United States)}]

**Comment by WheelDogs** [comment]
The problem is GitHub editor. It showed the backslashes immediately after the code was pasted. After leaving the editor, the \ were removed.

Went back to edit the script, hit the pen icon and the back slashes reappeared. tap someone that knows tcl or just add  a 'space, backslash' to the end of each line except the last.

**Comment by Naatan** [comment]
Please check the "code" tab here: https://guides.github.com/features/mastering-markdown/#examples

**Comment by cgchoffman** [comment]
Or just drag and drop a file in your comment.

**Comment by WheelDogs** [comment]
[kvars.txt](https://github.com/Komodo/KomodoEdit/files/759246/kvars.txt)

Seems someone just edited the text and the \ now appear...

**Comment by mitchell-as** [comment]
I loaded your code into Komodo IDE with ActiveTcl, started a debug session, and I'm getting accurate key values in the debug window. I see nothing like the screenshot in your original post. I am getting plain text values.

**Comment by Naatan** [comment]
![image](https://cloud.githubusercontent.com/assets/278649/22748814/555d84f8-edf9-11e6-972f-d8dfe5c06de7.png)

Am I missing something here? I gave you instructions on how to share code, seemed quite relevant seeing as you were having trouble sharing code.

**Comment by WheelDogs** [comment]
whew...let me try again...

**Comment by WheelDogs** [comment]
I get the same, reported result. Please advise on what I am doing wrong...
Please see attached..
![komodoarray](https://cloud.githubusercontent.com/assets/21134598/22767948/c87cb244-ee4c-11e6-847a-39a1e5802dde.png)

**Comment by WheelDogs** [comment]
Komodo IDE, version 10.1.4, build 89515, platform win32-x86.
Built on Fri Dec 16 10:34:21 2016.

**Comment by mitchell-as** [comment]
You are doing nothing wrong. It was my fault, since I know so little Tcl. Thanks for your infinite patience. This is a bug.
